### PR TITLE
Max's Amazing Multiple File Refactor

### PR DESCRIPTION
**Pull request by ianjennings**
- Multiple file refactor
- Linting
- Build
- Tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,20 +25,29 @@
...
+    "webpack": "^3.6.0",
+    "webpack-stream": "^4.0.0"
   },
   "dependencies": {
     "async": "^2.1.2",
     "axios": "^0.16.2",
     "eventemitter2": "^2.2.1",
-    "pubnub": "^4.13.0"
+    "pubnub": "^4.13.0",
+    "request": "^2.82.0"
```
Comment (ianjennings) [inline package.json]: I don't believe this is true. I had to move from ```request``` to ```axios```.

File: test/integration/main.test.js [hunk 2]
```diff
@@ -13,18 +13,18 @@ describe('import', function() {
 let me;
 let ChatEngine;
 let ChatEngineYou;
-let globalChannel  = 'thisistheglobalchannel-whatever' + new Date().getTime();
+let globalChannel = 'thisistheglobalchannel-whatever' + new Date().getTime();
 
-describe('config', function() {
+describe('config', () => {
```
Comment (ianjennings) [inline test/integration/main.test.js]: Mocha discourages arrow function.
https://mochajs.org/#arrow-functions

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            { test: /\.json/, loader: 'json' },
+        ],
+    },
+    output: {
+        filename: 'chat-engine.js',
+        library: 'ChatEngineCore',
+        libraryTarget: 'umd',
+    },
+    plugins: [
+        new StatsPlugin('stats.json', {
```
Comment (ianjennings) [inline webpack.config.js]: cool

File: src/bootstrap.js [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+
+    /**
+     * The {@link Chat} class.
+     * @member {Chat} Chat
+     * @memberof ChatEngine
+     * @see {@link Chat}
+     */
+    ChatEngine.Chat = class extends Chat {
+        constructor(...args) {
+            super(ChatEngine, ...args);
```
Comment (ianjennings) [inline src/bootstrap.js]: super baller solution

File: test/integration/main.test.js (hunk 2 above)
Comment (MaxPresman) [reply test/integration/main.test.js]: 👍 we can revert them back.

File: package.json (hunk 1 above)
Comment (MaxPresman) [reply package.json]: we need it on server.js : )

**Comment by ianjennings** [comment]
Ok I'll implement these changes and merge.

File: test/unit/modules/emitter.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+describe('#emitter', () => {
+    it('Emitter should be instanced', (done) => {
+        instance = new Emitter(Bootstrap({ publishKey: 'demo', subscribeKey: 'demo' }));
+        assert.isObject(instance, 'was successfully created');
+        done();
+    });
+
+    it('Emitter should be listened an event', (done) => {
+        instance.on('foo', (msg) => {
```
Comment (ianjennings) [inline test/unit/modules/emitter.test.js]: This test fails when running ```gulp unit_tests```

**Pull request merged**